test/AWS.Deploy.CLI.IntegrationTests/ServerMode/GetApplyOptionSettings.cs

```diff
@@ -144,48 +145,31 @@ await restClient.ApplyConfigSettingsAsync(sessionId, new ApplyConfigSettingsInpu
                 var setConfigResult = await restClient.ApplyConfigSettingsAsync(sessionId, new ApplyConfigSettingsInput()
                 {
                     UpdatedSettings = new Dictionary<string, string>()
                     {
-                        {"VPCConnector.Subnets", JsonConvert.SerializeObject(new List<string>{subnetsResources.Resources.Last().SystemName})},
-                        {"VPCConnector.SecurityGroups", JsonConvert.SerializeObject(new List<string>{securityGroupsResources.Resources.Last().SystemName})}
+                        {"VPCConnector.Subnets", JsonConvert.SerializeObject(new List<string>{subnet})},
+                        {"VPCConnector.SecurityGroups", JsonConvert.SerializeObject(new List<string>{securityGroup})}
                     }
                 });
 
-                await restClient.StartDeploymentAsync(sessionId);
-
-                await WaitForDeployment(restClient, sessionId);
-
-                var stackStatus = await _cloudFormationHelper.GetStackStatus(_stackName);
-                Assert.Equal(StackStatus.CREATE_COMPLETE, stackStatus);
-
-                Assert.True(logOutput.Length > 0);
-                Assert.Contains("Initiating deployment", logOutput.ToString());
-
-                var redeploymentSessionOutput = await restClient.StartDeploymentSessionAsync(new StartDeploymentSessionInput
-                {
-                    AwsRegion = _awsRegion,
-                    ProjectPath = projectPath
-                });
-
-                var redeploymentSessionId = redeploymentSessionOutput.SessionId;
+                var dryRunResponse = await restClient.StartDryRunDeploymentAsync(sessionId);
 
-                var existingDeployments = await restClient.GetExistingDeploymentsAsync(redeploymentSessionId);
-                var existingDeployment = existingDeployments.ExistingDeployments.First(x => string.Equals(_stackName, x.Name));
+                var metadata = await GetAppSettingsFromCFTemplate(dryRunResponse.CdkProject, _stackName);
 
-                Assert.Equal(_stackName, existingDeployment.Name);
-                Assert.Equal(appRunnerRecommendation.RecipeId, existingDeployment.RecipeId);
-                Assert.Equal(appRunnerRecommendation.Name, existingDeployment.RecipeName);
-                Assert.Equal(appRunnerRecommendation.ShortDescription, existingDeployment.ShortDescription);
-                Assert.Equal(appRunnerRecommendation.Description, existingDeployment.Description);
-                Assert.Equal(appRunnerRecommendation.TargetService, existingDeployment.TargetService);
-                Assert.Equal(DeploymentTypes.CloudFormationStack, existingDeployment.DeploymentType);
+                Assert.True(metadata.Settings.ContainsKey("VPCConnector"));
+                var vpcConnector = JsonConvert.DeserializeObject<VPCConnectorTypeHintResponse>(metadata.Settings["VPCConnector"].ToString());
```

Comment:
In simple terms yes, but the writing to JSON portion is done as part of the process of generating a CDK project and translating the Recommendation and setting configuration to a way the CDK project will understand. Then we are using CDK to compile the project and generate the CF template as it would during a normal deployment. Then we are inspecting the settings as part of the template. 
For the CDK assertion module you linked, I'm not quite sure how it would fit into this. Maybe we can have an offline discussion about it?

Reply:
Okay. Revisiting those docs, CDK's testing looks good for when you're in working in within the actual CDK project, but agree we're removed from that here.